src/Compilers/Core/Portable/DiagnosticAnalyzer/AnalyzerDriver.cs

```diff
@@ -845,35 +861,66 @@ private static Diagnostic GetFilteredDiagnostic(Diagnostic diagnostic, Compilati
             return compilation.Options.FilterDiagnostic(diagnostic);
         }
 
-        private static Task<AnalyzerActions> GetAnalyzerActionsAsync(
+        private static Task<Tuple<AnalyzerActions, ImmutableDictionary<DiagnosticAnalyzer, SemaphoreSlim>>> GetAnalyzerActionsAsync(
```

Comment:
I agree, but I wasn't sure a private method with a single callsite should introduce a new type wrapping the values.